server/app/services/migration/ProgramMigrationService.java

```diff
@@ -155,6 +156,51 @@ public ErrorAnd<ProgramMigrationWrapper, String> deserialize(
     }
   }
 
+  /**
+   * Question keys (names stripped of non-letter characters) in the imported JSON must be unique
+   * from one another AND if a question key exists in the DB, the question name must be exactly
+   * similar to that in the DB.
+   *
+   * @throws RuntimeException if there are any issues with the uniqueness of question keys
+   */
+  public void validateQuestionKeyUniqueness(ImmutableList<QuestionDefinition> questions) {
+    Map<String, ImmutableList<String>> questionKeysToNames = new HashMap<>();
+    for (QuestionDefinition question : questions) {
+      Optional<QuestionModel> conflictingQ = questionRepository.findConflictingQuestion(question);
+      if (conflictingQ.isPresent()
+          && !conflictingQ.get().getQuestionDefinition().getName().equals(question.getName())) {
+        throw new RuntimeException(
+            String.format(
+                "Question key %s (Admin ID \"%s\" stripped of non-letter characters) is already in"
+                    + " use by question %s. Please change the Admin ID so it either matches the"
+                    + " existing one, or compiles to a different key.",
```

Comment:
nit: I don't think "question key" is a user-facing phrase. I think we use `admin name` for the key, "question name" for the admin-facing name in the admin UI, and "question text" for the applicant-facing question text.